src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Withdraw.tsx

```diff
@@ -307,6 +322,60 @@ export const Withdraw: React.FC<WithdrawProps> = ({ accountId, onNext }) => {
     ],
   )
 
+  useEffect(() => {
+    if (!(accountId && inputValues && asset && opportunityData?.stakedAmountCryptoBaseUnit)) return
+    const { cryptoAmount } = inputValues
+    const amountCryptoBaseUnit = bnOrZero(cryptoAmount).times(bn(10).pow(asset.precision))
+
+    if (amountCryptoBaseUnit.isZero()) return
+
+    const debounced = debounce(async () => {
+      setQuoteLoading(true)
+      const withdrawBps = getWithdrawBps({
+        withdrawAmountCryptoBaseUnit: amountCryptoBaseUnit,
+        stakedAmountCryptoBaseUnit: opportunityData?.stakedAmountCryptoBaseUnit ?? '0',
+        rewardsamountCryptoBaseUnit: opportunityData?.rewardsAmountsCryptoBaseUnit?.[0] ?? '0',
+      })
+
+      const quote = await getThorchainSaversWithdrawQuote({ asset, accountId, bps: withdrawBps })
+      const { slippage_bps, expected_amount_out } = quote
+      const percentage = bnOrZero(slippage_bps).div(BASE_BPS_POINTS).times(100)
+
+      // total downside (slippage going into position) - 0.007 ETH for 5 ETH deposit
+      const cryptoSlippageAmountPrecision = bnOrZero(cryptoAmount).times(percentage).div(100)
+      setSlippageCryptoAmountPrecision(cryptoSlippageAmountPrecision.toString())
+
+      // daily upside
+      const dailyEarnAmount = bnOrZero(fromThorBaseUnit(expected_amount_out))
+        .times(opportunityData?.apy ?? 0)
+        .div(365)
+
+      const daysToBreakEven = cryptoSlippageAmountPrecision
+        .div(dailyEarnAmount)
+        .toFixed(0)
+        .toString()
+      setDaysToBreakEven(daysToBreakEven)
+      setQuoteLoading(false)
+    })
```

Comment:
Not needed in this PR but we might want to abstract the whole `daysToBreakEven` logic into a pure function since it gets reused across 3 places and is functionally the same